src/Agent.Worker/Release/ContainerFetchEngine/FetchEngine.cs

```diff
@@ -37,12 +37,21 @@ public abstract class FetchEngine : IDisposable
 
         public void Dispose()
         {
-            if (_downloadedFiles > 0)
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
why this? probably need a comment.

Reply:
I need to go back and see what the analyzer was telling me. I think, because this is abstract and IDisposable, then you need the other Dispose(bool) method to be virtual because Dispose() shouldn't be virtual? maybe? I followed the pattern I saw in other classes like this.